src/sentry/auth/access.py

```diff
@@ -957,10 +960,15 @@ def from_request_org_and_scopes(
             org_member=member,
         )
 
+        superuser_scopes = SUPERUSER_SCOPES
+        if not is_self_hosted() and features.has("auth:superuser-read-write", member):
+            if "superuser.write" not in auth_state.permissions:
```

Comment:
nit: could we combine the 2 ifs together? Seems like they all need to be invoked together for the single condition to happen right?